atlp-rwanda/eagles-ec-fe
# #187419133 Users should be able to chat

#### What does this PR do?

This PR Adds private and public chat functionality

#### Description of Task to be completed?

- a user should be able to read previous chats
- a user should be able to post a public chat
- users should be able to chat privately
- users should be able to receive messages in real time

#### How should this be manually tested?
1. Clone the repository.
2. Checkout to the branch `ft-chat-#187419133`.
3. Run `npm install` to install dependencies.
4. Run `npm run dev` to start the development server and login
5. navigate to `/chat`
6. Test if everything is working as expected by sending and receiving messages

#### What are the relevant pivotal tracker stories?

Delivers #187419133

### screenshots

Design

![MacBook Pro 14_ - 1](https://github.com/user-attachments/assets/ee41697a-194b-4a82-9f03-5dd42ae495d4)

Implementation

<img width="1680" alt="Screenshot 2024-07-22 at 4 27 03 PM" src="https://github.com/user-attachments/assets/c3fe0f99-677e-4cfe-ad2e-90ba64b6dc27">

<img width="1680" alt="Screenshot 2024-07-22 at 4 27 20 PM" src="https://github.com/user-attachments/assets/458cde59-e7e2-4a86-bcc7-c72abd48792a">

<img width="1680" alt="Screenshot 2024-07-22 at 4 27 49 PM" src="https://github.com/user-attachments/assets/47e0ed89-b7eb-46f9-a2eb-f9d2c37af969">
